cnd/src/swap_protocols/rfc003/alice/initiate_request.rs

```diff
@@ -0,0 +1,92 @@
+use crate::{
+    connector::Connector,
+    network::DialInformation,
+    swap_protocols::{
+        asset::Asset,
+        dependencies::LedgerEventDependencies,
+        rfc003::{
+            self,
+            alice::{spawner::AliceSpawn, SendRequest, State},
+            create_ledger_events::CreateLedgerEvents,
+            messages::ToRequest,
+            state_store::StateStore,
+            InsertState, Ledger,
+        },
+        Role, SwapId,
+    },
+};
+use futures_core::{
+    compat::Future01CompatExt,
+    future::{FutureExt, TryFutureExt},
+};
+use std::sync::Arc;
+
+pub trait InitiateRequest: Send + Sync + 'static {
+    fn initiate_request<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
+        &self,
+        id: SwapId,
+        bob_dial_info: DialInformation,
+        partial_swap_request: Box<dyn ToRequest<AL, BL, AA, BA>>,
+    ) -> Result<(), rfc003::insert_state::Error>
+    where
+        LedgerEventDependencies: CreateLedgerEvents<AL, AA> + CreateLedgerEvents<BL, BA>;
+}
+
+impl<S: SendRequest> InitiateRequest for Connector<S> {
+    fn initiate_request<AL: Ledger, BL: Ledger, AA: Asset, BA: Asset>(
+        &self,
+        id: SwapId,
+        bob_dial_info: DialInformation,
+        partial_swap_request: Box<dyn ToRequest<AL, BL, AA, BA>>,
+    ) -> Result<(), rfc003::insert_state::Error>
+    where
+        LedgerEventDependencies: CreateLedgerEvents<AL, AA> + CreateLedgerEvents<BL, BA>,
+    {
+        let swap_seed = self.deps.seed.swap_seed(id);
+        let swap_request = partial_swap_request.to_request(id, &swap_seed);
+
+        log::debug!("inserting initial state for alice");
```

Comment:
:thinking: Probably not a very useful log message without context.

Reply:
Woops - again.  This was just used during debugging - will remove.